lib/themes/dosomething/paraneue_dosomething/includes/preprocess/preprocess_node.inc

```diff
@@ -228,14 +227,23 @@ function paraneue_dosomething_preprocess_node_campaign(&$vars) {
     $vars['show_persistent_signup'] = theme_get_setting('show_persistent_signup');
   }
 
+  // Get prepared Campaign data.
+  try {
+    $current_campaign = Campaign::get($vars['nid']);
```

Comment:
Does this re-fetch the campaign from the database? If so, would it make sense to have a way to "hydrate" a Drupal node into a `Campaign` class without hitting the DB another time?